service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/SendEhrExtractCoreTaskExecutor.java

```diff
@@ -47,9 +49,12 @@ public void execute(SendEhrExtractCoreTaskDefinition sendEhrExtractCoreTaskDefin
 
         var request = mhsRequestBuilder.buildSendEhrExtractCoreRequest(
             stringRequestBody, sendEhrExtractCoreTaskDefinition.getConversationId(), sendEhrExtractCoreTaskDefinition.getFromOdsCode());
+        mhsClient.sendMessageToMHS(request);
         Instant requestSentAt = Instant.now();
-        mhsClient.sendEhrExtractCore(request, sendEhrExtractCoreTaskDefinition);
 
-        ehrExtractStatusService.updateEhrExtractStatusCore(sendEhrExtractCoreTaskDefinition, requestSentAt);
+        var ehrExtractStatus = ehrExtractStatusService.updateEhrExtractStatusCore(sendEhrExtractCoreTaskDefinition, requestSentAt);
+        if (ehrExtractStatus.getGpcAccessDocument().getDocuments().isEmpty()) {
+            sendAcknowledgementTaskDispatcher.send(ehrExtractStatus, POSITIVE_ACKNOWLEDGEMENT_TYPE_CODE);
```

Comment:
Consider:
```suggestion
            sendAcknowledgementTaskDispatcher.sendPositiveAcknowledgement(ehrExtractStatus);
```
more SOLID as SendEhrExtractCoreTaskExecutor does not need to know anything about how to send an acknowledgement